app/core/rate_limiting/generic_limiter.py

```
@@ -0,0 +1,531 @@
+        return await self.model.invoke(*args, **kwargs)
+    
+    def stream(self, *args, **kwargs) -> Any:
+        """Rate-limited streaming method."""
+        # For sync streaming, we do a simple check
+        estimated_tokens = self._estimate_tokens(args, kwargs)
+        
+        # Run async check in sync context using improved loop handling
+        try:
+            # Try to get existing event loop
+            try:
+                loop = asyncio.get_running_loop()
```

Ruff (0.12.2)

Local variable `loop` is assigned to but never used

Remove assignment to unused variable `loop`

(F841)